src/main/java/com/easypost/net/EasyPostResource.java

```diff
@@ -8,125 +8,20 @@
-import com.easypost.model.ErrorDeserializer;
-import com.easypost.model.Fee;
-import com.easypost.model.Shipment;
-import com.easypost.model.SmartrateCollection;
-import com.easypost.model.SmartrateCollectionDeserializer;
-import com.easypost.model.TrackingDetail;
-import com.google.gson.FieldNamingPolicy;
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
-import com.google.gson.JsonElement;
-import com.google.gson.JsonObject;
-
-import javax.net.ssl.HttpsURLConnection;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStream;
-import java.io.UnsupportedEncodingException;
-import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
-import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.lang.reflect.Modifier;
-import java.net.HttpURLConnection;
-import java.net.MalformedURLException;
-import java.net.URL;
-import java.net.URLEncoder;
-import java.net.URLStreamHandler;
+
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Date;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
-import java.util.Scanner;
-
-public abstract class EasyPostResource {
-    protected enum RequestMethod {
-        GET,
-        POST,
-        DELETE,
-        PUT
-    }
 
-    public static final String EASYPOST_SUPPORT_EMAIL = "support@easypost.com";
-    public static final Gson GSON =
-            new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
-                    .registerTypeAdapter(HashMap.class, new HashMapSerializer())
-                    .registerTypeAdapter(SmartrateCollection.class, new SmartrateCollectionDeserializer())
-                    .registerTypeAdapter(Error.class, new ErrorDeserializer()).create();
-    public static final Gson PRETTY_PRINT_GSON = new GsonBuilder().setPrettyPrinting().serializeNulls()
-            .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create();
-    public static final String CHARSET = "UTF-8";
-    public static final ArrayList<String> GLOBAL_FIELD_ACCESSORS =
-            new ArrayList<>(Arrays.asList("getCreatedAt", "getUpdatedAt", "getFees"));
-    public static final int DEFAULT_CONNECT_TIMEOUT_MILLISECONDS = 30000;
-    public static final int DEFAULT_READ_TIMEOUT_MILLISECONDS = 60000;
-    public static final double DEFAULT_APP_ENGINE_TIMEOUT_SECONDS = 20.0;
-    private static final String DNS_CACHE_TTL_PROPERTY_NAME = "networkaddress.cache.ttl";
-    private static final String CUSTOM_URL_STREAM_HANDLER_PROPERTY_NAME = "com.easypost.net.customURLStreamHandler";
-    private static int connectTimeoutMilliseconds = DEFAULT_CONNECT_TIMEOUT_MILLISECONDS;
-    private static int readTimeoutMilliseconds = DEFAULT_READ_TIMEOUT_MILLISECONDS;
-    private static double appEngineTimeoutSeconds = DEFAULT_APP_ENGINE_TIMEOUT_SECONDS;
+public abstract class EasyPostResource extends Requestor {
```

Comment:
I don't think this should extend Requestor. Requestor should be static HTTP logic that has nothing to do with EasyPost objects.

We may need to continue refactoring here. The problem with extending it is that all the downstream EasyPost resources will have access to whatever is available in Requestor which is not what we want.

Reply:
We do need to extend Requestor here, or else the classes will not be able to make any HTTP requests. 

This is how it looks like for the inheritance now. The subclass(e.g Address) -> EasyPostResource(which has the shared attributes that all classes have) -> Requestor(The HTTP request functions that all classes need to make API call). Basically, we have separated our HTTP requests function to a parent class of EasyPostResource

Reply:
We do not need to include it here. It's convenient that we do but there is no requirement.

My suggestion is we do not extend Requestor here and instead import the `request` function to these resource files explicitly. It will take a bit more refactoring to import a few more things but is probably the better option.

Reply:
Yes, we could do that. Actually, I should have moved `classURL()`, `instanceURL()` and enum `RequestMethod` to EasyPostResource. This will make only one import per file for the `request` function. I will address this next Monday.

Reply:
This `Requestor` is really the origins of the `Client` that we'll eventually make anyway. No object needs to extend `Requestor`, instead each object will internally have a configured `Requestor` instance that it will use to make HTTP calls.

Reply:
Resolved